### PR TITLE
(chore): Migrate CI to GitHub Actions

### PR DESCRIPTION
**Pull request by rashil2000**
The Appveyor tests here fail because of missing appveyor.yml file.

Once this is merged I will mirror this in Extras and Versions.

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rashil2000** [comment]
@niheaven is it possible to use caching here, similar to https://github.com/ScoopInstaller/Scoop/pull/4671?

**Comment by niheaven** [comment]
Yes, it should, since we have changelog.md and a release now. You should check the release to determine whether to use cache or download again.

Merge this and then investigate how to cache.

File: bin/test.ps1 [hunk 1]
```diff
@@ -1,4 +1,3 @@
-#requires -Modules @{ ModuleName = 'Pester'; ModuleVersion = '4.4.0' }
```
Comment (niheaven) [inline bin/test.ps1:1]: Why removes this line? Scoop's test hasn't updated to Pester 5...

```pwsh
#Requires -Modules @{ ModuleName = 'Pester'; MaximumVersion = '4.99' }
```

**Comment by rashil2000** [comment]
@niheaven this is good to go?

**Review by niheaven — APPROVED** [review]

**Comment by rashil2000** [comment]
@r15ch13 not sure why Appveyor is still running. Does it need to be disabled elsewhere?

**Comment by r15ch13** [comment]
yeah, I have to disable it in my appveyor account.

**Comment by rashil2000** [comment]
Okay, please disable it for Main, Extras and Versions

**Comment by r15ch13** [comment]
Done

**Pull request merged**